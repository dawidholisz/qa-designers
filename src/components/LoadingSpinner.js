import React from 'react'
import { CircleLoader } from 'react-spinners'

const LoadingSpinner = ({color}) => {
  return (
    <div className="loading-spinner">
      <CircleLoader size={150} sizeUnit={'px'} color={color} />
    </div>
  )
}

export default LoadingSpinner
