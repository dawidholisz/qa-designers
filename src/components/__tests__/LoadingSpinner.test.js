import React from 'react'
import { render } from '@testing-library/react'
import LoadingSpinner from '../LoadingSpinner'

const renderComponent = color => {
  return render(<LoadingSpinner color={color} />)
}

describe('<LoadningSpinner/>', () => {
  it('render white spinner', () => {
    const { container } = renderComponent('#FFF')
    expect(container.firstChild).toMatchSnapshot()
  })
  it('render black spinner', () => {
    const { container } = renderComponent('#000')
    expect(container.firstChild).toMatchSnapshot()
  })
})
