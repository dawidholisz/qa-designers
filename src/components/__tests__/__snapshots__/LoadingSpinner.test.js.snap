// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadningSpinner/> render black spinner 1`] = `
<div
  class="loading-spinner"
>
  <div
    class="css-1vsgq5g"
  >
    <div
      class="css-1no20o0"
    />
    <div
      class="css-u01n8e"
    />
    <div
      class="css-hvkkom"
    />
    <div
      class="css-4qvz6j"
    />
    <div
      class="css-lk2aob"
    />
  </div>
</div>
`;

exports[`<LoadningSpinner/> render white spinner 1`] = `
<div
  class="loading-spinner"
>
  <div
    class="css-1vsgq5g"
  >
    <div
      class="css-146hrkr"
    />
    <div
      class="css-b676xh"
    />
    <div
      class="css-aspt7i"
    />
    <div
      class="css-1w28y4f"
    />
    <div
      class="css-1hfeait"
    />
  </div>
</div>
`;
